packages/react-ui/src/app/features/connections/components/create-edit-connection-dialog-content.tsx

```diff
@@ -181,7 +181,14 @@ const CreateEditConnectionDialogContent = ({
         } else if (apError.code === ErrorCode.INVALID_APP_CONNECTION) {
           setErrorMessage(
             t('Connection failed with error {msg}', {
-              msg: apError.params.error,
+              msg:
+                typeof apError.params.error === 'string'
+                  ? apError.params.error
```

Comment:
Hi @ravikiranvm as we have multiple ternary expressions I would extract this to a function (maybe in a helpers file and add unit tests to this function, as the logic as I see becomes more complex.